### PR TITLE
[Task]: Removing deprecated getMasterRequest, in favor of getMainRequest

### LINKED ISSUE #12931: Remove RequestHelper::getMasterRequest
### Feature description

https://github.com/pimcore/pimcore/blob/c1f88e07434e56a1c04310cf6ec0588e8bce3687/lib/Http/RequestHelper.php#L107

### PR DESCRIPTION
**Pull request by kingjia90**
## Changes in this pull request  
Resolves #12931

### CODE REVIEW

**Comment by aryaantony92** [comment]
Shouldn't we replace the usage of hasMasterRequest() and getMasterRequest() from https://github.com/pimcore/demo/blob/9988bbb125cca2f7ce09e1002abef66d7bf09d63/src/Website/Tool/PimcoreUrl.php#L32
and https://github.com/pimcore/demo/blob/9988bbb125cca2f7ce09e1002abef66d7bf09d63/src/Website/Tool/PimcoreUrl.php#L33
also in demo and demo-enterprise ?

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 1]
```diff
@@ -131,10 +131,10 @@ Please use [event listener](../../20_Extending_Pimcore/13_Bundle_Developers_Guid
   ```
 - [Elements] Deprecated setting the parent id to 1 (root node), when an element with a non-existing parent id gets created.
 - [Custom Layouts] Deprecated Class `ClassLayoutDefinitionManager` and constant `PIMCORE_CUSTOMLAYOUT_DIRECTORY`.
+- [Request] Removed deprecated getMasterRequest() in favor of getMainRequest().
```
Comment (dvesh3) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: please move this to v11 upgrade notes

Comment (kingjia90) [reply doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]: ahh 🤦🏻‍♂️, thank you, nice catch

**Comment by kingjia90** [comment]
> Shouldn't we replace the usage of hasMasterRequest() and getMasterRequest() from https://github.com/pimcore/demo/blob/9988bbb125cca2f7ce09e1002abef66d7bf09d63/src/Website/Tool/PimcoreUrl.php#L32 and https://github.com/pimcore/demo/blob/9988bbb125cca2f7ce09e1002abef66d7bf09d63/src/Website/Tool/PimcoreUrl.php#L33 also in demo and demo-enterprise ?

Opened the PRs for Demo:
- [ ] https://github.com/pimcore/demo-enterprise/pull/52
- [ ] https://github.com/pimcore/demo/pull/353

**Pull request merged**